DemoApp/DemoAppCoordinator.swift

```diff
@@ -66,15 +66,23 @@ final class DemoAppCoordinator: NSObject, UNUserNotificationCenterDelegate {
 
     func presentChat(userCredentials: UserCredentials, channelID: ChannelId? = nil) {
         // Create a token
-        let token = try! Token(rawValue: userCredentials.token)
+        guard let token = try? Token(rawValue: userCredentials.token) else {
+            fatalError("There has been a problem getting the token, please check Stream API status")
```

Comment:
could we leverage [precondition failure over the fatalError](https://developer.apple.com/documentation/swift/1539374-preconditionfailure) here?
It's way safer way UX-wise for anybody getting across this issue.

Reply:
^^ @adamrushy

Reply:
I did think about using `preCondition` here @bielikb but I opted on the `fatalError` just because I think the unwrapping here is more readable inside the `guard` statement vs. a conditional check

Reply:
I hear you. Im not a fan of fatalErrors in general, but since this is a demo app, guess the “app crash” is understandable.